tools/parsecov/main.go

```diff
@@ -0,0 +1,125 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"log"
+	"os"
+	"path"
+	"sort"
+	"strconv"
+	"strings"
+
+	"gopkg.in/urfave/cli.v1"
+)
+
+var (
+	packagePrefixFlagName = "packagePrefix"
+)
+
+type count struct {
+	totalStatements   int
+	coveredStatements int
+}
+
+func (c count) percentage() float32 {
+	return 100 * float32(c.coveredStatements) / float32(c.totalStatements)
+}
+
+func parse(c *cli.Context) error {
+	packageCoverage := make(map[string]count)
+	profile, err := os.Open(c.Args()[0])
+	if err != nil {
+		return err
+	}
+	defer profile.Close()
+
+	// First line is "mode: foo", where foo is "set", "count", or "atomic".
+	// Rest of file is in the format
+	//	encoding/base64/base64.go:34.44,37.40 3 1
+	// where the fields are: name.go:line.column,line.column numberOfStatements count
+
+	// This program only handles set mode
+	s := bufio.NewScanner(profile)
+	s.Scan()
+	line := s.Text()
+	if line != "mode: set" {
+		return fmt.Errorf("invalid coverage mode, expecting 'mode: set' as the first line, but got %q", line)
+	}
+	linenum := 1
+	for s.Scan() {
+		line := s.Text()
+		linenum++
+		lastColon := strings.LastIndex(line, ":")
+		if lastColon == -1 {
+			return fmt.Errorf("line %d invalid: %q", linenum, line)
+		}
+
+		packageName := path.Dir(line[:lastColon])
+		numStatements, covered, err := getStats(line[lastColon+1:])
+		if err != nil {
+			return fmt.Errorf("line %d invalid: %q", linenum, line)
+		}
+		c := packageCoverage[packageName]
+		c.totalStatements += numStatements
+		if covered {
+			c.coveredStatements += numStatements
+		}
+		packageCoverage[packageName] = c
+	}
+
+	packageNames := make([]string, 0, len(packageCoverage))
+	for name := range packageCoverage {
+		packageNames = append(packageNames, name)
+	}
+	sort.Strings(packageNames)
+
+	var totalCount count
+	for _, v := range packageCoverage {
+		totalCount.totalStatements += v.totalStatements
+		totalCount.coveredStatements += v.coveredStatements
+	}
+	fmt.Printf("coverage: %5.1f%% of statements across all listed packages\n", totalCount.percentage())
```

Comment:
nit: consider a separate "PrintCoverage(map[name]count)" for all the output part

Reply:
Similar to my other comment, I think re-structuring at this point might end up being time wasted.